plugins/woocommerce/includes/admin/class-wc-admin-addons.php

```diff
@@ -87,17 +87,58 @@ public static function render_featured() {
 				)
 			);
 
-			if ( ! is_wp_error( $raw_featured ) ) {
-				$featured = json_decode( wp_remote_retrieve_body( $raw_featured ) );
-				if ( $featured ) {
-					set_transient( 'wc_addons_featured', $featured, DAY_IN_SECONDS );
+			if ( is_wp_error( $raw_featured ) ) {
+				do_action( 'woocommerce_page_wc-addons_connection_error', $raw_featured->get_error_message() );
+
+				$message = null;
+				if ( self::is_ssl_error( $raw_featured->get_error_message() ) ) {
+					$message = __( 'We encountered an SSL error. Please check if your site supports TLS version 1.2 or above.', 'woocommerce' );
 				}
+
+				self::output_empty( $message );
+
+				return;
+			}
+
+			$response_code = intval( wp_remote_retrieve_response_code( $raw_featured ) );
+			if ( 200 !== $response_code ) {
+				do_action( 'woocommerce_page_wc-addons_connection_error', $response_code );
+
+				/* translators: %d: HTTP error code. */
+				$message = sprintf(
+					esc_html(
+						__(
+							'Error code %d.',
+							'woocommerce'
+						)
+					),
+					$response_code
+				);
+
+				self::output_empty( $message );
+
+				return;
 			}
-		}
 
-		if ( ! empty( $featured ) ) {
-			self::output_featured( $featured );
+			$featured      = json_decode( wp_remote_retrieve_body( $raw_featured ) );
+			if ( empty( $featured ) || ! is_array( $featured ) ) {
+				do_action( 'woocommerce_page_wc-addons_connection_error', 'Empty or malformed response' );
+				$message = __( 'We received an empty or malformed response', 'woocommerce' );
+				self::output_empty( $message );
+
+				return;
+			}
+
+			if ( $featured ) {
+				set_transient( 'wc_addons_featured', $featured, DAY_IN_SECONDS );
+			}
 		}
+
+		self::output_featured( $featured );
+	}
+
+	public static function is_ssl_error( $error_message ) {
+		return ( false !== stripos( $error_message, 'cURL error 35' ) );
```

Comment:
Is it possible that there are other cURL codes for SSL errors that we want to capture? Or are we interested in only this specific one?

Reply:
We're mostly interested in catching this case, but I'm now outputting the cURL error in other cases too.